### ISSUE TITLE
Fork (5.0.X) with old loudness plugin

### ISSUE THREAD

**Issue by Mlocik97** [Mlocik97/pulseeffects#1]

**Comment by wwmm** [comment]
It is mostly a matter of looking at the commit that replaced this plugin by the one from LSP https://github.com/Mlocik97/pulseeffects/commit/fb0065dcbef2d9fee8d2aee1eb0477f9cd822975#diff-e214b36fc20743030c8e4fa15e4d604d17c5624cdbf635f81a34c8c6ebbc514f

Duplicate it of the current files first. Then rename them accordingly so the plugins do not conflict. After that it is just replacing the green lines in the commit by the old ones in pink.

**Comment by wwmm** [comment]
Ignore the file `src/sink_input_effects.cpp`

**Comment by Mlocik97** [comment]
Can I somehow have both new and old noise plugin in project?

**Comment by wwmm** [comment]
>Can I somehow have both new and old loudness plugin in project?

Yes. It is just a matter of duplicating some files with a different name. I still intend to do that in EasyEffects.  I just have to think on the best approach. The MDA loudness plugin was used through GStreamer's wrapper for LADSPA plugins. for now EasyEffects has only a minimal wrapper that I wrote for LV@ plugins. I have to consider if it is worth trying to write a wrapper for LADSPA too or if it is better to port the MDA plugin code to a Pipewire filter. That is assuming its license allows me to do so.

>Would it be possible to create last PulseEffects Legacy release?

I am not sure if I understand. Do you want a release for the legacy branch with the MDA plugin? Technically it can be done. I am not sure if it is the right decision as that branch is supposed to be in maintenance.

**Comment by Mlocik97** [comment]
OK I did some changes, not sure, I'm not good in C++, know only basics... so. I will create PR, and Would like if You can review it.

**Comment by wwmm** [comment]
Ok

**Comment by Mlocik97** [comment]
Thanks so much, I accidentaly renamed incorrect files, but it's fixed in PR now.

**Comment by wwmm** [comment]
I am not sure why I thought the MDA plugins used LADSPA. LV2 versions are available and they are probably the ones that were being used in PulseEffects before XD https://gitlab.com/drobilla/mda-lv2

Do you have a reason to want it in Pulseaudio? Now that I realized I can just use the LV2 Loudness plugin it should be very fast to add it to EasyEffects. Assuming you can use EasyEffects we would just waste effort trying to bring back the plugin to the legacy version.

**Comment by Mlocik97** [comment]
I use Ubuntu 20.04. so the only problem I see is GTK4, but I think it's possible to use it, also I think in Ubuntu 22.04. GTK4 will be already present... I was now searching on Internet and there is already some GTK4 package but it's not stable yet.

**Comment by wwmm** [comment]
The main problem is that gtk4 is not enough. You will need gtkmm4 too. That is the C++ wrapper for gtk4 we are using.

The maintainers of our flatpak package have already started to work in the EasyEffects package. So there is this to consider too.

**Comment by Mlocik97** [comment]
Hmm, looking on https://repology.org/project/gtkmm/versions and I see Fedora doesn't have gtkmm4 package, but Arch and Manjaro have.

**Comment by wwmm** [comment]
>I see Fedora doesn't have gtkmm4 package

Yes. Fedora's package maintainer is having a hard time because of this. I think the latest glibmm it is not evaiable there too.

Flatpak maintainers are also facing some difficulties but with the Meson version available in the platform runtime. The version there does not support c++20 =/
